crates/ruff_linter/resources/test/fixtures/flake8_bandit/S603.py

```diff
@@ -1,20 +1,34 @@
 from subprocess import Popen, call, check_call, check_output, run
 
 # Different Popen wrappers are checked.
-Popen("true", shell=False)
-call("true", shell=False)
-check_call("true", shell=False)
-check_output("true", shell=False)
-run("true", shell=False)
+a = input()
+Popen(a, shell=False)
+call(a, shell=False)
+check_call(a, shell=False)
+check_output(a, shell=False)
+run(a, shell=False)
 
 # Values that falsey values are treated as false.
-Popen("true", shell=0)
-Popen("true", shell=[])
-Popen("true", shell={})
-Popen("true", shell=None)
+Popen(a, shell=0)
+Popen(a, shell=[])
+Popen(a, shell={})
+Popen(a, shell=None)
 
 # Unknown values are treated as falsey.
-Popen("true", shell=True if True else False)
+Popen(a, shell=True if True else False)
 
 # No value is also caught.
-Popen("true")
+Popen(a)
+
+# Literals are fine, they're trusted.
+run("true")
+Popen(["true"])
+
+# Fine through assignments as well.
+run(c := "true")
+cmd = ["true"]
+run(cmd)
```

Comment:
I don't think I'd allow references because someone could call `cmd.push(inject)` and that would lead to a false positive. That's why I think we should only allow string literals and arrays of string literals

Reply:
Very fair, I'll disallow those then 👍

Reply:
Done, I chose to still allow `run(cmd := "true")` if that's fine though.